### PR TITLE
removed EnableJitInDiagMode from the code

### LINKED ISSUE #4311: Remove EnableJitInDiagMode from the code
It's an unused broken flag. We better remove it from the code and test cases. Long story short, if JIT is enabled, it's also enabled for Debugger and there is no another supported option.

### PR DESCRIPTION
**Pull request by mhaviv**
Hi @obastemur! Let me know if everything looks good to merge 🙂

Fixes #4311

### CODE REVIEW

**Comment by mhaviv** [comment]
Im guessing I did something wrong that some of the tests aren't passing... Can anyone let me know how to fix the remaining issues

**Comment by fatcerberus** [comment]
This seems to be the cause of the failures:
```
19:33:58 Linking CXX executable ch
19:34:09 Undefined symbols for architecture x86_64:
19:34:09   "JITTimeWorkItem::IsJitInDebugMode() const", referenced from:
19:34:09       Func::IsJitInDebugMode() const in libChakraCoreStatic.a(Func.cpp.o)
19:34:09       JITTimeWorkItem::InitializeReader(Js::ByteCodeReader*, Js::StatementReader<JsUtil::List<Js::FunctionBody::StatementMap*, Memory::ArenaAllocator, false, Js::CopyRemovePolicy, DefaultComparer> >*, Memory::ArenaAllocator*) in libChakraCoreStatic.a(JITTimeWorkItem.cpp.o)
19:34:09 ld: symbol(s) not found for architecture x86_64
```

**Comment by jackhorton** [comment]
Im guessing this fixes #4311 ?

**Comment by mhaviv** [comment]
hi @jackhorton yes it does 🙂(atleast its an attempt but not all the tests passed) 😶

File: lib/Jsrt/JsrtDebugManager.cpp [hunk 1]
```diff
@@ -452,11 +448,6 @@ void JsrtDebugManager::CallDebugEventCallbackForBreak(JsDiagDebugEvent debugEven
     {
         tempScriptContext->GetDebugContext()->GetProbeContainer()->AsyncDeactivate();
     }
-
-    if (Js::Configuration::Global.EnableJitInDebugMode())
-    {
-        scriptContext->GetThreadContext()->GetDebugManager()->GetDebuggingFlags()->SetForceInterpreter(false);
```
Comment (obastemur) [inline lib/Jsrt/JsrtDebugManager.cpp]: we need to either remove this target and use or leave this line as is.

File: lib/Jsrt/JsrtDebugManager.cpp [hunk 2]
```diff
@@ -369,10 +369,6 @@ bool JsrtDebugManager::EnableAsyncBreak(Js::ScriptContext* scriptContext)
     if (!probeContainer->IsAsyncActivate())
     {
         probeContainer->AsyncActivate(this);
-        if (Js::Configuration::Global.EnableJitInDebugMode())
-        {
-            scriptContext->GetThreadContext()->GetDebugManager()->GetDebuggingFlags()->SetForceInterpreter(true);
```
Comment (obastemur) [inline lib/Jsrt/JsrtDebugManager.cpp]: same here. read the comment below

**Comment by obastemur** [comment]
@mhaviv loved the effort!

About the changes; keep in mind that now JIT is always enabled during debug. So, all the conditions for `IsJitInDebugMode() == true`, we need to consider them as we may possibly need them by default.

**Comment by mhaviv** [comment]
Thank you so much for the feedback @obastemur I am going to fix it up now 👍

**Comment by rhuanjl** [comment]
@mhaviv your test failures are from not removing all uses of IsJitInDebugMode() - you've deleted the function but it has a few calls to it left in - they all need to go.

File: lib/Jsrt/JsrtDebugManager.cpp [hunk 3]
```diff
@@ -452,11 +448,6 @@ void JsrtDebugManager::CallDebugEventCallbackForBreak(JsDiagDebugEvent debugEven
     {
         tempScriptContext->GetDebugContext()->GetProbeContainer()->AsyncDeactivate();
     }
-
-    if (Js::Configuration::Global.EnableJitInDebugMode())
-    {
-        scriptContext->GetThreadContext()->GetDebugManager()->GetDebuggingFlags()->SetForceInterpreter(false);
-    }
 }
```
Comment (akroshg) [inline lib/Jsrt/JsrtDebugManager.cpp]: this function SetForceInterpreter, should be removed as well, I don't think anyone else is calling it other than debugger.

Comment (mhaviv) [reply lib/Jsrt/JsrtDebugManager.cpp]: Thank you @akroshg! I am going to remove it as well :)

File: README.md [hunk 4]
```diff
@@ -3,6 +3,10 @@
 [![Join the chat at https://gitter.im/Microsoft/ChakraCore](https://badges.gitter.im/Microsoft/ChakraCore.svg)](https://gitter.im/Microsoft/ChakraCore?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 [![Licensed under the MIT License](https://img.shields.io/badge/License-MIT-blue.svg)](https://github.com/Microsoft/ChakraCore/blob/master/LICENSE.txt)
 
+[![GitHub issues](https://img.shields.io/github/issues/Microsoft/ChakraCore.svg?&colorB=ff0000)](https://github.com/Microsoft/ChakraCore/issues)
+
+[![GitHub pull requests](https://img.shields.io/github/issues-pr/Microsoft/ChakraCore.svg?colorB=1FBF14)](https://github.com/Microsoft/ChakraCore/pulls)
+
```
Comment (dilijev) [inline README.md]: Not opposed to this change, but can this be a separate PR?

**Comment by obastemur** [comment]
@mhaviv I really want this PR to be going but It's a bit far away from the target. Are you planning to look into it?

**Comment by dilijev** [comment]
@mhaviv Just checking in -- are you planning to update this PR?

**Comment by mhaviv** [comment]
Hi @obastemur and @dilijev I'm really sorry for the delay!! I had alot going on for the past few weeks and I wasn't able to get to finish this issue. Hopefully I can finish it now but for some reason its still not passing the checks (I did the process all over again with a fresh copy of the repo).

**Comment by mhaviv** [comment]
Are there any parts of EnableJitInDebugMode() that I missed?

**Comment by fatcerberus** [comment]
@mhaviv You can click on "Details" next to any of the checks to see its build output and find out why you failed.  In this case it appears that you failed a bunch of unit tests simply because your code is triggering an assertion:
```
19:40:20 ASSERTION 39511: (/Users/dotnet-bot/j/workspace/Microsoft_ChakraCore/master/static_osx_osx_debug_prtest/lib/Backend/Func.cpp, line 789) m_nonTempLocalVars
19:40:20  Failure: (m_nonTempLocalVars)
```

Are you actually testing your changes?  You can run the tests locally, you don't have to push the changes upstream just to see if the unit tests pass.

**Comment by jackhorton** [comment]
hrm… I am surprised that --debug --test-build works. --debug makes a debug build (symbols, flags, no optimization) while --test-build makes a test build (symbols, flags, release optimization). You should only need one or the other. When you test it, youre running `python test\runtests.py`?

**Comment by dilijev** [comment]
@mhaviv To be clear: `--test-build` doesn't run tests, it refers to building in a configuration known as a "test build" which is more optimized than debug, but still allows you to run all tests. In order to run the tests you need to run the test script @jackhorton mentioned.

**Comment by dilijev** [comment]
@mhaviv are you planning to continue work on this PR?

**Comment by dilijev** [comment]
@dotnet-bot test this please

**Comment by dilijev** [comment]
@mhaviv are you planning to continue work on this PR?

**Comment by MikeHolman** [comment]
Closing due to lack of activity. @mhaviv, if you plan on continuing the PR, feel free to reopen.
